ray-operator/main.go

```diff
@@ -126,15 +131,25 @@ func main() {
 		LeaderElectionID:       "ray-operator-leader",
 	}
 
+	selectorsByObject, err := cacheSelectors()
```

Comment:
Currently the informer for batch Jobs resources watches all the Jobs, cluster-wide most of the time. This is not needed, as KubeRay is only interested in the Jobs it creates, and it can raise scalability concerns both for KubeRay (size of the informer cache) and the API server / etcd (watch events).

controller-runtime provide that mechanism to narrow the scope of the informer caches.

Reply:
What's the difference between 

```go
cache.MultiNamespacedCacheBuilder(watchNamespaces)
```

and 

```go
cache.MultiNamespacedCacheBuilder(watchNamespaces)(config, opts)
```
?

Reply:
I need to take a look at what is `SelectorsByObject` before I resolve this thread.

Reply:
Inlined as code comment. PTAL.

Reply:
> Currently the informer for batch Jobs resources watches all the Jobs, cluster-wide most of the time.

I want to double-check this statement. Are you saying that the informer still watches K8s Jobs from all namespaces even if we specify only one namespace in `watchNamespaces`?

Reply:
I meant that when the `watch-namespace` CLI option is not provided, which I assumed is the most general case, then the informer watches all the Kubernetes Jobs for all namespaces.